modules/EvseManager/EvseManager.cpp

```diff
@@ -397,6 +397,48 @@ void EvseManager::ready() {
                 }
             });
 
+            r_hlc[0]->subscribe_d20_dc_dynamic_charge_mode(
+                [this](types::iso15118_charger::DcChargeDynamicModeValues values) {
+                    constexpr auto PRE_CHARGE_MAX_POWER = 800.0f;
+
+                    bool target_changed{false};
+
+                    if (values.min_voltage > latest_target_voltage) {
+                        latest_target_voltage = values.min_voltage + 10; // TODO(sl): Check if okay
+                        target_changed = true;
+                    }
+                    if (values.max_voltage < latest_target_voltage) {
+                        latest_target_voltage = values.max_voltage - 10; // TODO(sl): Check if okay
+                        target_changed = true;
+                    }
+
+                    const double latest_target_power = latest_target_voltage * latest_target_current;
+
+                    if (latest_target_power <= PRE_CHARGE_MAX_POWER or values.min_charge_power > latest_target_power or
+                        values.max_charge_power < latest_target_power) {
```

Comment:
would do something like:
```
const auto latest_target_needs_update = [] (latest_target_voltage, latest_target_current) {
  constexpr static auto PRE_CHARGE_MAX_POWER = ...;
  const double power = .. x ...;
  return (power <= PRE_CH.. or ... or ..);
}

if (latest_target_needs_update(latest_target_voltage, latest_target_current) {
...
```
just in order to give some more appropriate name to the more complex if condition because it is just not easy to reason what this is doing